### PR TITLE
Persist gene search in distribution tab after login

### PR DESCRIPTION
**Pull request by bistline**
* Persisting gene search after a user logs in when clicking the 'Genome' tab 
* Bugfixes
  * Restoring javascript console in production environment
  * bin/job_monitor.rb correctly notices missing job workers that no longer have a pid file

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -15,13 +23,6 @@
       <%= render '/layouts/ga_script.js' %>
   </script>
 
-
-	<% if Rails.env == 'production' %>
-    <script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
```
Comment (eweitz) [inline app/views/layouts/application.html.erb]: Thanks for this!

File: app/views/site/view_gene_expression.html.erb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      </div>
+    </div>
+  </div>
+</div>
+
+<script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
+    // re-open a tab on reload if necessary
+    var urlParts = window.location.href.split('#');
+    if (urlParts.length > 1) {
+        openedTab = urlParts[1];
```
Comment (eweitz) [inline app/views/site/view_gene_expression.html.erb]:
```suggestion
        var openedTab = urlParts[1];
```

File: app/views/site/view_gene_expression_heatmap.html.erb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      </div>
+    </div>
+  </div>
+</div>
+
+<script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
+    // re-open a tab on reload if necessary
+    var urlParts = window.location.href.split('#');
+    if (urlParts.length > 1) {
+        openedTab = urlParts[1];
```
Comment (eweitz) [inline app/views/site/view_gene_expression_heatmap.html.erb]:
```suggestion
        var openedTab = urlParts[1];
```

File: app/views/site/view_gene_expression_heatmap.html.erb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        <div class="spinner-target" id="update-study-settings-spinner"></div>
+      </div>
+      <div class="modal-footer">
+        <button class="close" data-dismiss="modal">×</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
```
Comment (eweitz) [inline app/views/site/view_gene_expression_heatmap.html.erb]: I won't remember this code needs to be changed in two places, if that's ever needed.  

De-duplicating this code and [its other copy](https://github.com/broadinstitute/single_cell_portal_core/pull/250/files#diff-44ceb12e4c5a5d3a76b268ad3ab0c872R82) added here would fix that lurking maintainability issue.  If a quicker solution is desired, a code comment noting as much would also mitigate this.

Comment (bistline) [reply app/views/site/view_gene_expression_heatmap.html.erb]: Yes, you're correct.  Also, there's a much better way to write this function.  But I'll abstract it out and just make a function call so we're not duplicating code.

**Pull request merged**